kernel/src/net/socket/netlink/route/message/attr/link.rs

```diff
@@ -118,29 +118,52 @@ impl Attribute for LinkAttr {
         }
     }
 
-    fn read_from(reader: &mut dyn MultiRead) -> Result<Self>
+    fn read_from(header: &CAttrHeader, reader: &mut dyn MultiRead) -> Result<Option<Self>>
     where
         Self: Sized,
     {
-        let header = reader.read_val_opt::<CAttrHeader>()?.unwrap();
-
         // TODO: Currently, `IS_NET_BYTEORDER_MASK` and `IS_NESTED_MASK` are ignored.
-        let res = match LinkAttrClass::try_from(header.type_())? {
-            LinkAttrClass::IFNAME => Self::Name(reader.read_cstring_with_max_len(IFNAME_SIZE)?),
-            LinkAttrClass::MTU => Self::Mtu(reader.read_val_opt()?.unwrap()),
-            LinkAttrClass::TXQLEN => Self::TxqLen(reader.read_val_opt()?.unwrap()),
-            LinkAttrClass::LINKMODE => Self::LinkMode(reader.read_val_opt()?.unwrap()),
-            LinkAttrClass::EXT_MASK => Self::ExtMask(reader.read_val_opt()?.unwrap()),
-            class => {
-                // FIXME: Netlink should ignore all unknown attributes.
-                // But how to decide the payload type if the class is unknown?
-                // Reference: https://docs.kernel.org/userspace-api/netlink/intro.html#unknown-attributes
+        let class = LinkAttrClass::try_from(header.type_());
```

Comment:
How about handling the error case directly here? This way, the subsequent match arms can be simplified, and we won't need to include the Ok handling in each match arm.

```rust 
let payload_len = header.payload_len();
let Ok(class) = LinkAttrClass::try_from(header.type_()) else {
    // Unknown attributes should be ignored.
    // Reference: <https://docs.kernel.org/userspace-api/netlink/intro.html#unknown-attributes>.
    reader.skip_some(payload_len);
    return Ok(None);
}
```

Reply:
If so, we'll have to repeat the following code twice:
```
reader.skip_some(payload_len);
return Ok(None);
```

It's not a big deal, though.